File: src/main/java/pl/yalgrin/playnite/simplesync/mapper/RegionMapper.java
package pl.yalgrin.playnite.simplesync.mapper;

import org.apache.commons.lang3.StringUtils;
import org.springframework.stereotype.Component;
import pl.yalgrin.playnite.simplesync.domain.Region;
import pl.yalgrin.playnite.simplesync.dto.RegionDTO;

@Component
public class RegionMapper extends AbstractObjectMapper<Region, RegionDTO> {

    @Override
    protected void fillOtherFields(Region entity, RegionDTO dto) {
        entity.setSpecificationId(dto.getSpecificationId());
    }

    @Override
    protected void fillDtoFields(RegionDTO dto, Region entity) {
        dto.setSpecificationId(entity.getSpecificationId());
    }

    @Override
    protected boolean hasChanged(RegionDTO dto, Region target) {
        return super.hasChanged(dto, target) || !StringUtils.equals(target.getSpecificationId(),
                dto.getSpecificationId());
    }

    @Override
    protected RegionDTO createDTO() {
        return new RegionDTO();
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/mapper/RegionMapper.kt
package pl.yalgrin.playnite.simplesync.library.mapper

import org.apache.commons.lang3.Strings
import org.springframework.stereotype.Component
import pl.yalgrin.playnite.simplesync.library.domain.Region
import pl.yalgrin.playnite.simplesync.library.dto.RegionDTO
import reactor.core.publisher.Mono

@Component
class RegionMapper : LibraryObjectMapperImpl<Region, RegionDTO>() {
    override fun fillOtherFields(
        entity: Region, dto: RegionDTO
    ): Mono<Region> {
        return Mono.fromSupplier {
            entity.specificationId = dto.specificationId
            entity
        }
    }

    override fun fillOtherDtoFields(
        dto: RegionDTO, entity: Region
    ): Mono<RegionDTO> {
        return Mono.fromSupplier {
            dto.specificationId = entity.specificationId
            dto
        }
    }

    override fun hasChanged(
        dto: RegionDTO, target: Region
    ): Boolean {
        return super.hasChanged(dto, target) || !Strings.CS.equals(
            target.specificationId, dto.specificationId
        )
    }

    override fun createDTO(): RegionDTO = RegionDTO()
}